http.ts

```diff
@@ -121,8 +125,80 @@ export class ServerRequest {
   method: string;
   proto: string;
   headers: Headers;
+  conn: Conn;
+  r: BufReader;
   w: BufWriter;
 
+  public async *readChunk() {
+    if (this.headers.has("content-length")) {
+      const len = +this.headers.get("content-length");
+      if (Number.isNaN(len)) {
+        return new Uint8Array(0);
+      }
+      let buf = new Uint8Array(1024);
+      let rr = await this.r.read(buf);
+      let nread = rr.nread;
+      while (!rr.eof && nread < len) {
+        yield buf.subarray(0, rr.nread);
+        buf = new Uint8Array(1024);
+        rr = await this.r.read(buf);
+        nread += rr.nread;
+      }
+      yield buf.subarray(0, rr.nread);
+    } else if (this.headers.get("transfer-encoding") === "chunked") {
```

Comment:
`Transfer-Encoding` might be comma separated list: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding. At the moment we don't handle any other encoding, but probably it's worth to add a TODO

Reply:
Sure. (Missed this due to referring to an old RFC)